server/sonar-server/src/test/java/org/sonar/server/view/index/ViewIndexerTest.java

```diff
@@ -204,4 +260,20 @@ private ComponentDto addProjectWithIssue(RuleDto rule, OrganizationDto org) {
     return project;
   }
 
+  private void doRecover(BooleanSupplier condition) throws InterruptedException {
+    MapSettings settings = new MapSettings()
+      .setProperty("sonar.search.recovery.initialDelayInMs", "0")
+      .setProperty("sonar.search.recovery.minAgeInMs", "1")
+      .setProperty("sonar.search.recovery.delayInMs", "1");
+
+    RecoveryIndexer recoveryIndexer = new RecoveryIndexer(System2.INSTANCE, settings.asConfig(), dbClient, underTest);
+    recoveryIndexer.start();
+
+    // Wait for recovery
+    while (condition.getAsBoolean()) {
+      Thread.sleep(1_000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2925)